org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMTextFormatter.java

```diff
@@ -68,8 +72,13 @@ public void formatText(DOMText textNode, XMLFormattingConstraints parentConstrai
 							insertLineBreak(spaceStart, contentStart, edits);
 							availableLineWidth = getMaxLineWidth();
 						}
-					} else {
+					} else if (isJoinContentLines()
+							|| !isJoinContentLines() && (spaceStart == textNode.getStart() || !containsNewLine)) {

```

Comment:
no need to check `!isJoinContentLines()` in the second part of the condition, you can do just `isJoinContentLines() || (spaceStart == textNode.getStart() || !containsNewLine)`